cluster-autoscaler/processors/status/scale_down_status_processor.go

```diff
@@ -61,6 +61,34 @@ const (
 	ScaleDownInProgress
 )
 
+// NodeDeleteResultType denotes the type of the result of node deletion. It provides deeper
+// insight into why the node failed to be deleted.
+type NodeDeleteResultType int
+
+const (
+	// NodeDeleteOk - the node was deleted successfully.
+	NodeDeleteOk NodeDeleteResultType = iota
+
+	// NodeDeleteErrorFailedToMarkToBeDeleted - node deletion failed because the node couldn't be marked to be deleted.
+	NodeDeleteErrorFailedToMarkToBeDeleted
+	// NodeDeleteErrorFailedToEvictPods - node deletion failed because some of the pods couldn't be evicted from the node.
+	NodeDeleteErrorFailedToEvictPods
+	// NodeDeleteErrorPodsFailedToDisappear - node deletion failed because some of the pods didn't disappear from the node in time after eviction.
+	NodeDeleteErrorPodsFailedToDisappear
```

Comment:
Added error information

Reply:
I meant do we even need 2 types - it feels like it could be covered by one (like "NodeDeleteErrorFailedToEvictPods" or "NodeDeleteErrorFailedDrain" or sth) ;)

Reply:
Agreed, only one left!